src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -16,7 +17,7 @@
      * @param type - any class of Machine sub class. For example: Truck.class or Bulldozer.class
      * @return the list of machines
      */
-    List<Object> getAll(Class type);
+    List<T> getAll(Class<? extends Machine> type);
```

Comment:
Consider creating a local variable in your implementation of `getAll(Class type)` method to improve code readability and avoid direct operations on the method's return value.